### DISCUSSION TITLE
Adding tags to `CfnCrawler` results in invalid tag input

### DISCUSSION THREAD

**Discussion by juanchaves** [aws/aws-cdk#30992] [Q&A] [answered]
I think I am missing something as everything I try to tag this CfnCrawler gives back: 
```
Invalid tag input expected map of {key: value} have [{"key":"testKey","value":"testValue"}]                    
      at MapFormatter.parseTags (node_modules/aws-cdk-lib/core/lib/tag-manager.js:1:1918)
      at TagManager.parseExternalTags (node_modules/aws-cdk-lib/core/lib/tag-manager.js:1:5608)
      at new TagManager (node_modules/aws-cdk-lib/core/lib/tag-manager.js:1:3980)
      at new CfnCrawler (node_modules/aws-cdk-lib/aws-glue/lib/glue.generated.js:1:27809)
      ...
``` 

As far as I can tell, in the CDK they're using `TagManager` for this property and the type that is listed [in the docs](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-glue-crawler.html#cfn-glue-crawler-tags) is Array of [Tag](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-resource-tags.html#w43aac23c14d498c13c15).

Notes: 
- the reason I am using `.forEach` is that this is a custom Construct that I am writing for other users and they might supply an array of multiple tags
- the `TagProps` interface below comes from a common library that we use. The class that uses this interface uses `Aspects` to add some mandatory tags.

### scenario 1:

```typescript
export interface TagProps {
  /**
   *key as string
   */
  readonly key: string
  /**
   * value of tag as string
   */
  readonly value: string
}

const tagProps: TagProps[] = [{key: "testKey", value: "testValue"}]

const crawler: glue.CfnCrawler = new glue.CfnCrawler(this, `Crawler`, {
    ...crawler props
    tags: tagProps
})

```

Result: `Invalid tag input expected map of {key: value} have [{"key":"testKey","value":"testValue"}]`

### scenario 2:

```typescript
... TagProps interface

const tagProps: TagProps[] = [{key: "testKey", value: "testValue"}]

const crawler: glue.CfnCrawler = new glue.CfnCrawler(this, `Crawler`, {
    ...crawler props
    tags: undefined
})

// Trying to use the CDK Tag class
tagProps.forEach((tag: TagProps) => new Tag(tag.key, tag.value, { includeResourceTypes: ['CfnCrawler']}))

```

Also tried using the `Tag` class without the `cdk.TagProp` `{ includeResourceTypes: ['CfnCrawler']}`

Result: `Invalid tag input expected map of {key: value} have [{"key":"testKey","value":"testValue"}]`

### scenario 3:

```typescript
... TagProps interface

const tagProps: TagProps[] = [{key: "testKey", value: "testValue"}]

const crawler: glue.CfnCrawler = new glue.CfnCrawler(this, `Crawler`, {
    ...crawler props
    tags: undefined
})

// Trying to use the CDK Tag class
tagProps.forEach((tag: TagProps) => Tags.of(crawler).add(tag.key, tag.value, { includeResourceTypes: ['CfnCrawler']}))

```

Also tried using the `Tags` class without the `cdk.TagProp` `{ includeResourceTypes: ['CfnCrawler']}`

Result: `Invalid tag input expected map of {key: value} have [{"key":"testKey","value":"testValue"}]`

### scenario 4:

```typescript
... TagProps interface

const crawler: glue.CfnCrawler = new glue.CfnCrawler(this, `Crawler`, {
      ...crawler props

      tags: undefined,
})

const tagProps: TagProps[] = [{key: "testKey", value: "testValue"}]
// Trying to use the exact "type" as the CfnCrawler as per Docs
let crawlerTags: {"Key": string, "Value": string}[] = []
tagProps.forEach((tag: commonLib.TagProps) => crawlerTags.push({"Key": tag.key, "Value": tag.value}))
const cfnCrawler = (crawler.node.defaultChild as CfnCrawler)
cfnCrawler.addPropertyOverride('Tags', crawlerTags)
```

Result: `Invalid tag input expected map of {key: value} have [{"key":"testKey","value":"testValue"}]`

### scenario 5:

```typescript
... TagProps interface

const tagProps: TagProps[] = [{key: "testKey", value: "testValue"}]

const crawler: glue.CfnCrawler = new glue.CfnCrawler(this, `Crawler`, {
    ...crawler props
    tags: tagProps ? new Map(tagProps.map(object => { return [object.key, object.value] })) : undefined,
})
```

Result: `Invalid tag input expected map of {key: value} have [{"key":"testKey","value":"testValue"}]`

I would appreciate any help in this 😄

**Comment by Rizxcviii** [comment, marked as answer]
Hi,
I had the same issue you did, but managed to get around it by using the `setTag` method by doing `crawler.tags.setTag('key','value')`
